### PR TITLE
[SPARK-21625][DOC] Add incompatible Hive UDF describe to DOC

### PR DESCRIPTION
**Pull request by wangyum**
## What changes were proposed in this pull request?

Add incompatible Hive UDF describe to DOC.

## How was this patch tested?

N/A

### CODE REVIEW

**Comment by maropu** [comment]
Pg does not accept this query;
```
postgres=# select sqrt(3);
       sqrt       
------------------
 1.73205080756888
(1 row)

postgres=# select sqrt(-1);
ERROR:  cannot take square root of a negative number
```
So, Another solution is to fail in analysis.

**Comment by hvanhovell** [comment]
@maropu that only works for literals. I am sort-of in favor of the Hive default; it seems kinda bad to bring down a job because of negative value.

**Comment by maropu** [comment]
@hvanhovell yea, both is okay to me. just suggestion.

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/MathExpressionsSuite.scala [hunk 1]
```diff
@@ -403,11 +403,13 @@ class MathExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
 
   test("sqrt") {
     testUnary(Sqrt, math.sqrt, (0 to 20).map(_ * 0.1))
-    testUnary(Sqrt, math.sqrt, (-5 to -1).map(_ * 1.0), expectNaN = true)
+    testUnary(Sqrt, math.sqrt, (-5 to -1).map(_ * 1.0), expectNull = true)
```
Comment (viirya) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/MathExpressionsSuite.scala]: Looks like you're changing the NaN cases for many math expressions to Null. I'm not sure if we can do thing like this to break compatibility.

Comment (HyukjinKwon) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/MathExpressionsSuite.scala]: Yea, I was writing this comment. If `NaN` makes sense in a way, I was thinking we couldn't consider this case as a bug that should be fixed.

Comment (wangyum) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/MathExpressionsSuite.scala]: Yes, we migrated Hive and MySQL SQLs to Spark, and found some inconsistencies. `NaN` is unfamiliar to MySQL and Oracle users

Comment (wangyum) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/MathExpressionsSuite.scala]: Users always use `is not null` filter invalid value, but spark sql breaks:
```
         > create table spark_21625 as select 10 as c1, sqrt(-10) as c2;
spark-sql> select * from spark_21625;
10      NaN                                                                     
spark-sql> select * from spark_21625 where c2 is not null;
10	NaN
```

Comment (viirya) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/MathExpressionsSuite.scala]: We have `IsNaN`. So users might already use it to check those invalid values.

**Comment by gatorsmile** [comment]
Can we document this difference in https://spark.apache.org/docs/latest/sql-programming-guide.html#compatibility-with-apache-hive?

**Comment by gatorsmile** [comment]
retest this please

File: docs/sql-programming-guide.md [hunk 2]
```diff
@@ -1957,6 +1957,14 @@ Not all the APIs of the Hive UDF/UDTF/UDAF are supported by Spark SQL. Below are
   Spark SQL currently does not support the reuse of aggregation.
 * `getWindowingEvaluator` (`GenericUDAFEvaluator`) is a function to optimize aggregation by evaluating
   an aggregate over a fixed window.
+  
+### Incompatible Hive UDF
+
+Below is a list of Hive UDFs that we don't incompatible yet.
```
Comment (gatorsmile) [inline docs/sql-programming-guide.md]: ? Below are the scenarios in which Hive and Spark generate different results

**Comment by gatorsmile** [comment]
Thanks! Merged to master.

**Pull request closed without merging**